src/app/gm-tools/util/event-form.ts

```diff
@@ -35,7 +35,7 @@ export interface EventForm {
 
 export function eventFormFactory(
   saveMethod: (
-    event: OmitAutoGeneratedColumns<EventModel>,
+    event: Partial<OmitAutoGeneratedColumns<EventModel>>,
```

Comment:
This code could use `Omit<T>` to be more specific

https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys

```typescript
Omit<OmitAutoGeneratedColumns<EventModel>, 'round_number'>
```

Reply:
Actually, looking at this again, maybe `Omit` doesn't quite fit this use case. Because we will have a `round_number` _sometimes_. Not never.

I assume what we want is to have `round_number` be an optional property. TypeScript doesn't have a built in way to create a type where specified properties are optional, but I found [this StackOverflow post](https://stackoverflow.com/questions/43159887/make-a-single-property-optional-in-typescript) that can do it.

```typescript
type Optional<T, K extends keyof T> = Pick<Partial<T>, K> & Omit<T, K>;
```

Or you could just ignore all this and merge if it works lol

Reply:
Toyed around with this for a bit. The eventual cast to a regular OAGC<EventModel> complained that they didn't overlap. I don't think it's particularly worth spending more time on.